### ISSUE TITLE
Please kindly update for Cordova 3.x. Awesome plugin btw

### ISSUE THREAD

**Issue by donniemceduns** [DrewDahlman/ImageFilter#4]

**Comment by quanghoc** [comment]
1+

Have anyone tried to use Javascript filter like CamanJS to accomplish this?

**Comment by billmn** [comment]
@quanghoc Yes but with Cordova seems to be really slow (like 15 seconda to apply a filter)

**Comment by quanghoc** [comment]
@billmn I'm thinking about rewriting CamanJS in an async / non-blocking way. Current bad performance is mostly due to the WebObject. iPhone 5S took about 1-2 sec but still noticable. What I did was to resize the image before processing.

**Comment by billmn** [comment]
@quanghoc Have you seen this? https://github.com/meltingice/CamanJS/wiki/About-the-block-renderer

**Comment by travisdahl** [comment]
+1 would love to use this!

**Comment by quanghoc** [comment]
@billmn I haven't tested block render. Did it make things faster?

**Comment by billmn** [comment]
Benefits:
- Image rendering is asynchronous, which means page rendering is no longer blocked.
- Slower filters get a nice speed improvement due to (fake) render concurrency. Of course browser JS engines are event-based and single-threaded, but by splitting up the canvas into separate chunks, we give more control to the JS engine for managing operations.

Caman already use this but is really slow yet.

I think the only way to have good performance in image processing is to take advantage of GPU ... therefore need a plugin that allows native access to hardware

**Comment by DrewDahlman** [comment]
Before writing this I tried caman and js rendering. The speed and quality will always lack from native.

**Comment by travisdahl** [comment]
Agreed @DrewDahlman . I just tried it, and although its cool for the web it leaves a lot to be desired on the phone. So Drew, from one Dahl-man to another, what are your thoughts on updating this? eh? ;p

**Comment by mohamedsalehamin** [comment]
any news about updating that
